sherpa/astro/data.py

```diff
@@ -4892,7 +4892,53 @@ def unsubtract(self):
 
 
 class DataIMG(Data2D):
-    "Image data set, including functions for coordinate transformations"
+    '''Image data set, including functions for coordinate transformations
+
+    Parameters
+    ----------
+    name : string
+        name of this dataset
+    x0 : array-like
+        Coordinates in one dimension
+    x1 : array-like
+        Coordinates in the other dimension
+    y : array-like
+        The values of the dependent observable.
+    shape : tuple
+        Shape of the data grid for regularly gridded data (optional). This is
+        used return the data as an image e.g. for display,
+        but it not needed for fitting and modelling.
+    staterror : array-like
+        the statistical error associated with the data
+    syserror : array-like
+        the systematic error associated with the data
+    sky : `sherpa.astro.io.wcs.WCS`
+        The optional WCS object that converts to the physical coordinate system.
+    eqpos : `sherpa.astro.io.wcs.WCS`
+        The optional WCS object that converts to the world coordinate system.
+    coord : string
+        The coordinate system of the data. This can be one of 'logical',
+        'physical', or 'world'.
+        'image' is an alias for 'logical' and 'wcs' is an alias for 'world'.
+    header : dict
+        The FITS header associated with the data to store meta data.
+
+    Examples
+    --------
+    In this example, we create an image data set, but do not specify the
+    coordinate system. This means that the coordinates are expressed in
+    the logical coordinates of the image, i.e. in pixels::
+
+        >>> from sherpa.astro.data import DataIMG
+        >>> import numpy as np
+        >>> x1, x0 = np.mgrid[20:30, 5:20]
+        >>> datashape = x0.shape
+        >>> y = np.sqrt((x0 - 10)**2 + (x1 - 31)**2)
+        >>> x0 = x0.flatten()
+        >>> x1 = x1.flatten()
+        >>> y = y.flatten()
+        >>> image = DataIMG("bimage", x0=x0, x1=x1, y=y, shape=datashape)
+    '''
```

Comment:
There's a few things to think about here (and also in the `DataIMGInt` case you mention below). They don't need to all (or maybe any) addressed here, but I think we should have some understanding of the issues. Some of the following are highly interlinked (or can be read as "you've already said this"):

- technically this example is not the "logical" coordinate system as CIAO thinks of it, because that starts at 1,1 (defining the middle of the first pixel) following the FITS standard.
- however, in practice it only really matters when we have set up the `sky` and/or `eqpos` transforms, which we don't here (but I claim we do actually want to understand what me mean, as I discuss in the last point)
- It's not immediately obvious to me what `DataIMG` adds compared to `Data2D` for this particular use case (there are definitely differences, including the notebook auto-display code, as well as when sending data to DS9, it's just I can't enumerate them)
- on a related note, I struggle to understand why/when we need to send in the `shape` argument
- can we come up with an understanding of when using `DataIMG` is more appropriate than `Data2D`?
- coming back to this example, and the question you raise about the `DataIMGInt` example: in #1764 I have tried to ask the question, but probably not directly enough, of 
  - what do we mean by the x0 and x1 axis coordinates and how does that interact with any `coord` argument oyuo give to the constructor (because, if not careful, we have problems like #1762)
  - what happens when we don't have a transform

Reply:
There is some text in `Data2D` and `Data2DInt` that suggests why you might prefer to use `DataIMG` / `DataIMGInt` which might better be here (or in both places)?

Reply:
That text does not apply here. As you said in a different location, I can use `DataIMG` for sparse data. There is just no benefit in the `DataIMG` in that case, it simply acts the same way as the parent class. In contrast, if I have anon-sparse image, the parent class still doesn ot give me any of the non-sparse functionality, so in that case you DO want to use a different (i.e. this) class.

Reply:
To the list of questions:

- I agree that all that needs to be resolved, but not in this PR. Here, I'm simply documenting what the current input paramters hare. Some of that is ambiguous, precisely because we do not have the answer to those questions. In those cases, I've kept the description in the docstring short; it can be expanded when we have better answers, but at least a user (i.e. me when I opened that PR) knows hwere to start and what the arguments are called.
- Technically, what I'm putting in is the "physical" coordinate system (numbers from 5-20 and 20-30). And what exactly is supposed (or does) happen wit hcoordinates at that point is fuzzy, so this is one of those locations where I say little and hope that we can expand the docstring once we resolve the coordinate issues.
- I woudl argue that `DataIMG` is more appropriate if you (a) have multiple coordinate systems (e.g. image, physcial, sky), or (b) want regions filters, or (c) niceties like notebook display for non-sparse images.
-